packages/block-library/src/image/image.js

```diff
@@ -333,21 +328,6 @@ export default function Image( {
 		} );
 	}
 
-	function updateAlignment( nextAlign ) {
-		const extraUpdatedAttributes = [ 'wide', 'full' ].includes( nextAlign )
-			? {
-					width: undefined,
-					height: undefined,
-					aspectRatio: undefined,
-					scale: undefined,
-			  }
-			: {};
```

Comment:
As @fabiankaegy mentioned, resetting these values seems to be the one custom thing the image block did when changing alignments, and it should probably be retained.

An effect would be one way to handle it, the downside being modifying alignment and resetting the size become two separate changes on the undo stack.

Reply:
Would it work to still handle it in an effect but using the markNextChangeAsNonPersistent method to not create anything in the undo stack for the change?

Not sure whether the change would properly get reverted when someone does go back one step at that point. I just know we have something for that